targets/TARGET_TOSHIBA/TARGET_TMPM46B/spi_api.c

```diff
@@ -0,0 +1,295 @@
+/* mbed Microcontroller Library
+ *******************************************************************************
+ * (C)Copyright TOSHIBA ELECTRONIC DEVICES & STORAGE CORPORATION 2017 All rights reserved
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 3. Neither the name of STMicroelectronics nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************
+ */
+#include "spi_api.h"
+#include "mbed_error.h"
+#include "pinmap.h"
+#include "tmpm46b_ssp.h"
+
+static const PinMap PinMap_SPI_SCLK[] = {
+    {PK4, SPI_0, PIN_DATA(2, 1)},
+    {PF3, SPI_1, PIN_DATA(5, 1)},
+    {PD3, SPI_2, PIN_DATA(1, 1)},
+    {NC,  NC,    0}
+};
+
+static const PinMap PinMap_SPI_MOSI[] = {
+    {PK3, SPI_0, PIN_DATA(2, 1)},
+    {PF4, SPI_1, PIN_DATA(5, 1)},
+    {PD2, SPI_2, PIN_DATA(1, 1)},
+    {NC,  NC,    0}
+};
+
+static const PinMap PinMap_SPI_MISO[] = {
+    {PK2, SPI_0, PIN_DATA(2, 0)},
+    {PF5, SPI_1, PIN_DATA(5, 0)},
+    {PD1, SPI_2, PIN_DATA(1, 0)},
+    {NC,  NC,    0}
+};
+
+static const PinMap PinMap_SPI_SSEL[] = {
+    {PK1, SPI_0, PIN_DATA(2, 1)},
+    {PF6, SPI_1, PIN_DATA(5, 1)},
+    {PD0, SPI_2, PIN_DATA(1, 1)},
+    {NC,  NC,    0}
+};
+
+typedef enum {
+    SPI_SPH0_SPO0 = 0,   // [Clock Phase 0, Clock Polarity 0] = 0
+    SPI_SPH0_SPO1,       // [Clock Phase 0, Clock Polarity 1] = 1
+    SPI_SPH1_SPO0,       // [Clock Phase 1, Clock Polarity 0] = 2
+    SPI_SPH1_SPO1        // [Clock Phase 1, Clock Polarity 1] = 3
+} SPI_MODE;
```

Comment:
this is confusing, one enum is `SPI_MODE` and right below `spi_mode`.

Reply:
@0xc0170, removed these enums.